### ISSUE TITLE
UEFI + Win10Home + VeraCrypt 1.23: UEFI does not offer any options

### ISSUE THREAD

**Issue by DeVIL-I386** [veracrypt/VeraCrypt#400]
After my system partition has been encrypted, no more options are displayed in the UEFI. 

No UEFI update has been installed in the meantime.

![20190117_010449561](https://user-images.githubusercontent.com/1786308/51286978-b708cd00-19f5-11e9-86aa-8f70637b48cc.png)

![20181202_161321230](https://user-images.githubusercontent.com/1786308/51287060-0818c100-19f6-11e9-8429-7549bcf19959.png)

![zwischenablage01](https://user-images.githubusercontent.com/1786308/51287031-f2a39700-19f5-11e9-937e-996a68733933.png)

**Comment by idrassi** [comment]
This is the first time I see such problem when only the system partition partition is encrypted. Is it possible to know the model of your PC?

Obviously, EFI firmware settings should not be affected by the content of a partition of the disk since the user can choose to format the disk completely and install another OS like Linux.

Did you encrypt another partition other than the system partition?

Anyway, I don’t think we can do anything for this issue since VeraCrypt follows standard practice for EFI configuration.

**Comment by DeVIL-I386** [comment]
Notebook is a: Acer Swift 1 (SF114-32)

Only the Windows partition was encrypted.

![zwischenablage02](https://user-images.githubusercontent.com/1786308/51287592-8aa28000-19f8-11e9-8b08-979368230dc6.png)

**Comment by DeVIL-I386** [comment]
Decrypted the system partition.

![zwischenablage04](https://user-images.githubusercontent.com/1786308/51289455-8e3a0500-1a00-11e9-9c94-8d3be5729896.png)

![zwischenablage05](https://user-images.githubusercontent.com/1786308/51289463-942fe600-1a00-11e9-974d-a24254522ab3.png)

![20190117_022744045](https://user-images.githubusercontent.com/1786308/51289444-7febe900-1a00-11e9-9d55-73ed85f4f4ba.png)

**Comment by idrassi** [comment]
This confirms that the EFI firmware needs to access the content of the system partition.

This is very suspicious and if I were you I will not use this machine to store sensitive information.

Thank you for reporting this, it is a good information for other Acer users.

**Comment by DeVIL-I386** [comment]
My device was manufactured on 03.09.2018 and comes with "BIOS"-Version 1.03.

On 08.01.2019 the manufacturer has released version 1.07.
[https://www.acer.com/ac/de/DE/content/support-product/7564?b=1&pn=NX.GXUEG.001](https://www.acer.com/ac/de/DE/content/support-product/7564?b=1&pn=NX.GXUEG.001)

I will install this version now and then encrypt the system partition with VeraCrypt 1.23 again.

**Comment by DeVIL-I386** [comment]
![imag3475](https://user-images.githubusercontent.com/1786308/51351922-6e601b00-1aac-11e9-9798-55323f24cea6.png)
![20190117_225552589](https://user-images.githubusercontent.com/1786308/51351938-7c15a080-1aac-11e9-90c1-4c24941b3994.png)

Will start encrypting in a few minutes.

**Comment by DeVIL-I386** [comment]
![imag3480](https://user-images.githubusercontent.com/1786308/51356090-999d3700-1ab9-11e9-8b88-5ebef658fe5b.jpg)

In the old BIOS/UEFI version 1.03 "VeraCrypt BootLoader (DcsBoot)" was displayed where now "<null string[](url)>" is written.
After I moved the entry (VeraCrypt) from position 2 to position 1 yesterday, I had the problem with UEFI. Before that I had access to all settings in UEFI.

I had the idea to move the entry because I read that Windows 10 likes to repair the UEFI entries and then you have problems decrypting the partition.
It was recommended to set a UEFI administrator password, but I couldn't find a way.

Now with BIOS/UEFI version 1.07 I haven't moved the entry yet.

**Comment by meekstone** [comment]
I'm not using VeraCrypt but I faced a similar issue with the same hardware (Acer Swift 1 SF114-32) and BIOS version 1.09 and found a workaround.

Any unsigned UEFI file seems to be displayed as `<null string>` in the BIOS (but is correctly labelled in the F12 boot menu). Putting such a `<null string>` item as first item in the "Boot priority order" in the BIOS renders the BIOS inaccessible, but the system will still boot properly (and use this `<null string>` as the first boot option).

I followed the steps below to configure a workaround (assuming a BIOS with default options, so some steps may not be required for you). Also, I installed Debian, which required disabling "secure boot", which may not be applicable to your case.

1. Set a "supervisor password"
2. Disable "secure boot"
3. Install operating system (or VeraCrypt) and its UEFI file
4. Reenable "secure boot"
5. Use "Security -> Select an UEFI file as trusted for executing" to add the UEFI file as trusted
6. Save changes and exit
7. Enter BIOS again and put the trusted UEFI file (which is now also listed as boot option) as first boot option
8. Disable "secure boot"
9. The BIOS is now still accessible and the UEFI file of your choice is the first boot option

In short: add VeraCrypt's UEFI file as trusted, then put this trusted UEFI file as first boot option.

For me it seems that the BIOS does not need access to the system partition. I think the BIOS parses the "trusted UEFI files database/list" and UEFI files not in this database/list cause a problem when set as first boot option, probably due to missing a "name".

I contacted Acer regarding this issue and will post any useful information here, should Acer provide such information in a timely manner.

**Comment by meekstone** [comment]
Acer's response is that adding the UEFI file to the "trusted UEFI files" is the only supported way to have another OS installed in parallel to Windows and to have said OS' bootloader set as primary boot option.

Which means Acer won't provide a BIOS which fixes the issue of an inaccessible BIOS as Acer does not see this as an issue.

**Comment by groundstack** [comment]
Hello, I have the same laptop, SF114-32. My issue is different in the sense I can not access the BIOS menu at all, I just see a non-blinking cursor in the top left. I also can not boot from USB and boot menu also does not work. Totally bricked. 

In my system I removed Windows completely, then installed Void linux.

At the time right before it became unusable I was changing boot order with 'efibootmgr'

My question, those UEFI settings changed by 'efibootmgr' reside in eMMC or SO8 flash chip that holds UEFI firmware? Does the BIOS need a Windows installation on eMMC to work?

Any help appreciated, sending back to Acer is not an option as I am overseas. Thanks for any tips

**Comment by DeVIL-I386** [comment]
There is currently no solution to the problem.
